cwf/gateway/docker/c/Dockerfile

```diff
@@ -15,65 +15,48 @@
 # Builder image for C binaries and Magma proto files
 # -----------------------------------------------------------------------------
 
-# debian stretch image
-# ARG OS_DIST=debian
-# ARG OS_RELEASE=stretch
-# ARG EXTRA_REPO=https://facebookconnectivity.jfrog.io/artifactory/list/dev/
-# ARG CLANG_VERSION=3.8
-
 # ubuntu bionic image
 ARG OS_DIST=ubuntu
 ARG OS_RELEASE=bionic
 ARG EXTRA_REPO=https://facebookconnectivity.jfrog.io/artifactory/cwf-prod
-ARG CLANG_VERSION=3.9
 
 # Stretch is required for c build
 FROM $OS_DIST:$OS_RELEASE AS builder
 ARG OS_DIST
 ARG OS_RELEASE
 ARG EXTRA_REPO
-ARG CLANG_VERSION
 
 # Add the magma apt repo
 RUN apt-get update && \
-    apt-get install -y apt-utils software-properties-common apt-transport-https gnupg
+    apt-get install -y apt-utils software-properties-common apt-transport-https gnupg wget
 COPY orc8r/tools/ansible/roles/pkgrepo/files/jfrog.pub /tmp/jfrog.pub
 
 RUN apt-key add /tmp/jfrog.pub && \
     apt-add-repository "deb ${EXTRA_REPO} ${OS_RELEASE} main"
 
+# Install bazel
+WORKDIR /usr/sbin
+RUN wget --progress=dot:giga https://github.com/bazelbuild/bazelisk/releases/download/v1.10.0/bazelisk-linux-amd64 && \
+    chmod +x bazelisk-linux-amd64 && \
+    ln -s /usr/sbin/bazelisk-linux-amd64 /usr/sbin/bazel
 
 # Install dependencies required for building
 RUN apt-get -y update && apt-get -y install \
```

Comment:
If you are looking to optimize build time (the 1 minute difference seems okay to me, but if you are looking for areas to improvise), eliminating the additional apt-get update commands could be something to consider.

Reply:
On second thoughts, considering docker uses a cache, future builds *may not* perform an update if a previous unchanged layer already had it. Let's leave the updates intact for every install, but see if they can be combined by any means.